functions/utils/analytics.js

```diff
@@ -0,0 +1,43 @@
+const functions = require("firebase-functions");
+const { db, admin } = require("./admin");
+
+/*
+    writeToAnalytics adds an event trigger to 'applicants' collection.
+    The event trigger updates a separate document's fields in the 'analytics' collection.
+    Fields include: 
+                -> total number of applicants
+                -> number of UCSC student applicants
+                -> number of first time hackers
+    **NOTE**
+    Fields are not mutually exclusive!
+    i.e. in the event of a document creation;
+            -> if an applicant were a first time hacker and a UCSC student:
+                --> ALL THREE FIELDS WILL BE INCREMENTED BY ONE
+                    
+*/
+
+const service = functions.firestore.document("applicants/{docId}").onCreate(async (snap) => {
+  const document = snap.data();
+  const analyticsDoc = db.collection("analytics").doc("applicant-analytics");
+  const toUpdate = {
+    ucscIncrement: document.ucscStudent ? 1 : 0,
+    firstTimeIncrement: document.firstCruzHack ? 1 : 0,
```

Comment:
We might want to change this in the future depending on how we store the data, whether it's by school name/or isUCSC  but for now this is good!